nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1991,11 +1991,11 @@ private boolean mayBeNullFieldAccess(VisitorState state, ExpressionTree expr, Sy
 
   private Description matchDereference(
       ExpressionTree baseExpression, ExpressionTree derefExpression, VisitorState state) {
-    Symbol dereferenced = ASTHelpers.getSymbol(baseExpression);
-    if (dereferenced == null
-        || dereferenced.type.isPrimitive()
-        || dereferenced.getKind() == ElementKind.PACKAGE
-        || ElementUtils.isTypeElement(dereferenced)) {
+    Symbol baseExpressionSymbol = ASTHelpers.getSymbol(baseExpression);
+    if (baseExpressionSymbol != null
+        && (baseExpressionSymbol.type.isPrimitive()
```

Comment:
Maybe worth making this logic two nested ifs for clarity sake, and also adding a quick comment about in which case `.getSymbol(...)` might return `null` and yet there could be a null derefence? (Obvious example of ternary expressions in the PR, but long term I am not sure I'd remember it)

Reply:
f178408